source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,620 @@
+#else
+#define FSL_ENET_BUFF_ALIGNMENT MAX(ENET_BUFF_ALIGNMENT, FSL_FEATURE_L2CACHE_LINESIZE_BYTE)
+#endif
+#elif defined(FSL_FEATURE_L1DCACHE_LINESIZE_BYTE)
+#define FSL_ENET_BUFF_ALIGNMENT MAX(ENET_BUFF_ALIGNMENT, FSL_FEATURE_L1DCACHE_LINESIZE_BYTE)
+#else
+#define FSL_ENET_BUFF_ALIGNMENT ENET_BUFF_ALIGNMENT
+#endif
+#else
+#define FSL_ENET_BUFF_ALIGNMENT ENET_BUFF_ALIGNMENT
+#endif
+
+typedef uint8_t rx_buffer_t[SDK_SIZEALIGN(ENET_RXBUFF_SIZE, FSL_ENET_BUFF_ALIGNMENT)];
+typedef uint8_t tx_buffer_t[SDK_SIZEALIGN(ENET_TXBUFF_SIZE, FSL_ENET_BUFF_ALIGNMENT)];
+
+/*!
+ * @brief Used to wrap received data in a pbuf to be passed into lwIP
+ *        without copying.
+ * Once last reference is released, buffer can be used by ENET RX DMA again.
+ */
+typedef struct rx_pbuf_wrapper
+{
+    void *buffer;              /*!< Original buffer wrapped. */
+    volatile bool buffer_used; /*!< Wrapped buffer is used by ENET or FreeRTOS+TCP. */
+} rx_pbuf_wrapper_t;
+
+/**
+ * Helper structure to hold private data used to operate your Ethernet interface.
+ */
+struct ethernetif
+{
+    ENET_Type *base;
+    enet_handle_t handle;
+    enet_rx_bd_struct_t *RxBuffDescrip;
+    enet_tx_bd_struct_t *TxBuffDescrip;
+    rx_buffer_t *RxDataBuff;
+    tx_buffer_t *TxDataBuff;
+    rx_pbuf_wrapper_t RxPbufs[ENET_RXBUFF_NUM];
+};
+
+
+static mdio_handle_t mdioHandle = { .ops = &EXAMPLE_MDIO_OPS };
+
+static phy_handle_t phyHandle = { .phyAddr = 0x00, .mdioHandle = &mdioHandle, .ops = &EXAMPLE_PHY_OPS };
+
+/*-----------------------------------------------------------*/
+
+/*
+ * A deferred interrupt handler task that processes
+ */
+
+static TaskHandle_t receiveTaskHandle;
+
+static struct ethernetif EthernetInterface1;
+struct ethernetif * ethernetifLocal = &EthernetInterface1;
+
+/*-----------------------------------------------------------*/
+
+/* The queue used to communicate Ethernet events with the IP task. */
+extern QueueHandle_t xNetworkEventQueue;
+
+/*-----------------------------------------------------------*/
+
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_rx_bd_struct_t rxBuffDescrip_0[ ENET_RXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_tx_bd_struct_t txBuffDescrip_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static rx_buffer_t rxDataBuff_0[ ENET_RXBUFF_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static tx_buffer_t txDataBuff_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+
+static void prvEMACHandlerTask( void * pvParameters );
+
+static void ethernet_callback( ENET_Type *base,
+                               enet_handle_t *handle,
+                               enet_event_t event,
+                               enet_frame_info_t *frameInfo,
+                               void *userData );
+
+static void prvProcessFrame( int length );
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+	status_t xStatus;
+	phy_config_t xConfig = { 0 };
+	BaseType_t xResult = pdPASS;
```

Comment:
I would choose `pdFAIL` as a default value, because only one branch leads to `pdPASS`.

Reply:
Makes sense, fixed.